### ISSUE TITLE
Non-string keys error

### ISSUE THREAD

**Issue by vorner** [dtolnay/serde-ignored#5]
Hello

I'm getting a non-string key error when trying to deserialize things like `HashMap<usize, usize>`. How hard would it be to support them?

This panics on the last command:

```rust
use std::collections::HashMap;

use serde_yaml::Value;

fn main() {
    const CFG: &str = r#"
    1: 42
    2: 0
    "#;
    let x: HashMap<usize, usize> = serde_yaml::from_str(CFG).unwrap();
    let d: Value = serde_yaml::from_str(CFG).unwrap();
    let y: HashMap<usize, usize> = serde_ignored::deserialize(d, |p| {
        println!("Ignoring {}", p);
    }).unwrap();
}
```

**Comment by dtolnay** [comment]
Fixed in 0.1.0.

**Comment by vorner** [comment]
Thank you! :-)
